AllReadyApp/Web-App/AllReady/Areas/Admin/ViewModels/Validators/Task/TaskEditViewModelValidator.cs

```diff
@@ -1,68 +1,68 @@
+    {
+        private readonly IMediator _mediator;
+
+        public TaskEditViewModelValidator(IMediator mediator)
+        {
+            if (mediator == null)
+            {
+                throw new ArgumentNullException(nameof(mediator));
+            }
+
+            _mediator = mediator;
+        }
+
+        public async Task<List<KeyValuePair<string, string>>> Validate(EditViewModel viewModel)
+        {
+            var result = new List<KeyValuePair<string, string>>();
+
             var parentEvent = await _mediator.SendAsync(new EventByEventIdQuery { EventId = viewModel.EventId });
-
-
-            // Rule - End date cannot be earlier than start date
-            if (viewModel.EndDateTime < viewModel.StartDateTime)
-            {
-                result.Add(new KeyValuePair<string, string>(nameof(viewModel.EndDateTime), "End date cannot be earlier than the start date"));
-            }
-
-            // Rule - Start date cannot be out of range of parent event
-            if (viewModel.StartDateTime < parentEvent.StartDateTime)
-            {
-                result.Add(new KeyValuePair<string, string>(nameof(viewModel.StartDateTime), "Start date cannot be earlier than the event start date " + parentEvent.StartDateTime.ToString("d")));
-            }
-
-            // Rule - End date cannot be out of range of parent event
-            if (viewModel.EndDateTime > parentEvent.EndDateTime)
-            {
-                result.Add(new KeyValuePair<string, string>(nameof(viewModel.EndDateTime), "End date cannot be later than the event end date " + parentEvent.EndDateTime.ToString("d")));
-            }
-
-            // Rule - Itinerary tasks must start and end on same calendar day
-            if (parentEvent.EventType == EventType.Itinerary)
-            {
-                if (viewModel.StartDateTime.Date != viewModel.EndDateTime.Date)
-                {
-                    result.Add(new KeyValuePair<string, string>(nameof(viewModel.EndDateTime), "For itinerary events the task end date must occur on the same day as the start date. Tasks cannot span multiple days"));
-                }
-            }
-
-            return result;
-        }
-    }
-
-    public interface ITaskEditViewModelValidator
-    {
+
+
+            // Rule - End date cannot be earlier than start date
+            if (viewModel.EndDateTime < viewModel.StartDateTime)
+            {
+                result.Add(new KeyValuePair<string, string>(nameof(viewModel.EndDateTime), "End date cannot be earlier than the start date"));
+            }
+
+            // Rule - Start date cannot be out of range of parent event
+            if (viewModel.StartDateTime < parentEvent.StartDateTime)
+            {
+                result.Add(new KeyValuePair<string, string>(nameof(viewModel.StartDateTime), "Start date cannot be earlier than the event start date " + parentEvent.StartDateTime.ToString("f")));

```

Comment:
We have typically been using "g" as the date/time pattern.
Also prefer using string.Format or string interpolation instead of concatenating strings. for example:
$"Start date cannot be earlier than the event start date {parentEvent.StartDateTime.ToString("g")}"